streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -213,6 +212,31 @@ public <K, V> void send(final String topic,
                     keyClass,
                     valueClass),
                 exception);
+        } catch (final SerializationException exception) {
```

Comment:
As pointed out during the KIP discussion, the new handler should also cover `ClassCastException` -- we only had a special catch-block for it, to get a better error message (we still want a good error message for this case, but we should also apply the handler and allow people to CONTINUE on a ClassCastException).